tests/test_source_gcp_secret_manager.py

```diff
@@ -0,0 +1,212 @@
+
+    client.parse_secret_path = SecretManagerServiceClient.parse_secret_path
+
+    def mock_list_secrets(parent: str) -> list:
+        # Mock list_secrets
+        secret = mocker.Mock()
+        secret.name = f'{parent}/secrets/test-secret'
+        return [secret]
+
+    client.list_secrets = mock_list_secrets
+
+    secret_response = mocker.Mock()
+    secret_response.payload.data.decode.return_value = 'test-value'
+
+    def mock_access_secret_version(name: str):
+        if name == 'projects/test-project/secrets/test-secret/versions/latest':
+            return secret_response
+        else:
+            raise KeyError()
+
+    client.access_secret_version = mock_access_secret_version
+
+    return client
+
+
+@pytest.fixture
+def secret_manager_mapping(mock_secret_client):
+    return GoogleSecretManagerMapping(mock_secret_client, 'test-project')
+
+
+@pytest.fixture
+def test_settings():
+    class TestSettings(BaseSettings):
+        test_secret: str
+        another_secret: str
+
+    return TestSettings
+
+
+@pytest.fixture(autouse=True)
+def mock_google_auth(mocker: MockerFixture):
+    mocker.patch(
+        'pydantic_settings.sources.providers.gcp.google_auth_default', return_value=(mocker.Mock(), 'default-project')
+    )
+
+
+def test_secret_manager_mapping_init(secret_manager_mapping):
+    assert secret_manager_mapping._project_id == 'test-project'
+    assert len(secret_manager_mapping._loaded_secrets) == 0
+
+
+def test_secret_manager_no_project_raises_attribute_error(secret_manager_mapping):
+    assert secret_manager_mapping._project_id == 'test-project'
+    assert len(secret_manager_mapping._loaded_secrets) == 0
+
+
+def test_secret_manager_mapping_gcp_project_path(secret_manager_mapping, mock_secret_client):
+    secret_manager_mapping._gcp_project_path
+    mock_secret_client.common_project_path.assert_called_once_with('test-project')
+
+
+def test_secret_manager_mapping_secret_names(secret_manager_mapping):
+    names = secret_manager_mapping._secret_names
+    assert names == ['test-secret']
+
+
+def test_secret_manager_mapping_getitem_success(secret_manager_mapping):
+    value = secret_manager_mapping['test-secret']
+    assert value == 'test-value'
+
+
+def test_secret_manager_mapping_getitem_nonexistent_key(secret_manager_mapping):
+    with pytest.raises(KeyError):
+        _ = secret_manager_mapping['nonexistent-secret']
+
+
+def test_secret_manager_mapping_getitem_access_error(secret_manager_mapping, mocker):
+    secret_manager_mapping._secret_client.access_secret_version = mocker.Mock(side_effect=Exception('Access denied'))
+
+    with pytest.raises(KeyError):
+        _ = secret_manager_mapping['test-secret']
+
+
+def test_secret_manager_mapping_len(secret_manager_mapping):
+    assert len(secret_manager_mapping) == 1
+
+
+def test_secret_manager_mapping_iter(secret_manager_mapping):
+    assert list(secret_manager_mapping) == ['test-secret']
```

Comment:
These two tests can be merged in one. with `assert list(secret_manager_mapping) == ['test-secret']` , `assert len(secret_manager_mapping) == 1` is redundant